node/bft/src/helpers/round_cache.rs

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::{ensure, Result};
+use snarkvm::{
+    console::types::{Address, Field},
+    ledger::committee::Committee,
+    prelude::Network,
+};
+use std::collections::{HashSet, VecDeque};
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+struct AddressCoordinates<N: Network>((Field<N>, Field<N>));
+
+impl<N: Network> AddressCoordinates<N> {
+    fn from_address(address: Address<N>) -> Self {
+        let xy = address.to_group().to_xy_coordinates();
+        Self((xy.0, xy.1))
+    }
+}
+
+impl<N: Network> Ord for AddressCoordinates<N> {
+    /// Compares two AddressCoordinates
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match self.0.0 == other.0.0 {
+            true => self.0.1.cmp(&other.0.1),
+            false => self.0.0.cmp(&other.0.0),
+        }
+    }
+}
+
+impl<N: Network> PartialOrd for AddressCoordinates<N> {
+    /// Compares two AddressCoordinates
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[derive(Debug)]
+pub struct RoundCache<N: Network> {
+    /// The current highest round which has (stake-weighted) quorum
+    last_highest_round_with_quorum: u64,
+    /// A list of (round, Vec<Address>), indicating the last seen highest round for each address
```

Comment:
```suggestion
    /// A list of (round, Vec<AddressCoordinates>), indicating the last seen highest round for each address coordinates
```

Reply:
Changed and fixed in [9159ede](https://github.com/AleoHQ/snarkOS/pull/2897/commits/9159ede39ecf9141fb35e753ad0df3c94ff9aaf3)